### PR TITLE
Set up Comment Front end

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #314 :*
https://github.com/bcgov/met-public/issues/#314

Create Comment list and api end point

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-web/src/routes/AuthenticatedRoutes.tsx [hunk 1]
```diff
@@ -19,6 +20,7 @@ const AuthenticatedRoutes = () => {
             <Route path="/survey/submit/:surveyId" element={<SurveySubmit />} />
             <Route path="/engagement/form/:engagementId" element={<EngagementForm />} />
             <Route path="/engagement/view/:engagementId" element={<EngagementView />} />
+             <Route path="/comments" element={<CommentListing />} />
```
Comment (djnunez-aot) [inline met-web/src/routes/AuthenticatedRoutes.tsx:23]: surveyid

File: met-web/src/services/commentService/index.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import http from 'apiManager/httpRequestHandler';
+import { Comment } from 'models/comment';
+import Endpoints from 'apiManager/endpoints';
+import { replaceAllInURL, replaceUrl } from 'helper';
+
+interface FetchCommentParams {
+    unlinked?: boolean;
```
Comment (djnunez-aot) [inline met-web/src/services/commentService/index.tsx:7]: surveyid

File: met-web/src/services/commentService/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if (response.data.result) {
+            return Promise.resolve(response.data.result);
+        }
+        return Promise.reject(response.data.message ?? 'Failed to fetch comment');
+    } catch (err) {
+        return Promise.reject(err);
+    }
+};
+
+interface PostCommentRequest {
```
Comment (djnunez-aot) [inline met-web/src/services/commentService/index.tsx:30]: remove

File: met-web/src/services/commentService/index.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    name: string;
+    components?: unknown[];
+    engagement_id?: string;
+    form_json: unknown;
+}
+export const postComment = async (data: PostCommentRequest): Promise<Comment> => {
```
Comment (djnunez-aot) [inline met-web/src/services/commentService/index.tsx:36]: remove

File: met-web/src/services/commentService/index.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    } catch (err) {
+        return Promise.reject(err);
+    }
+};
+
+interface LinkPutCommentRequest {
+    id: string;
+    engagement_id: string;
+}
+export const linkComment = async (params: LinkPutCommentRequest): Promise<Comment> => {
```
Comment (djnunez-aot) [inline met-web/src/services/commentService/index.tsx:68]: remove

File: met-web/src/apiManager/endpoints/index.ts [hunk 6]
```diff
@@ -23,6 +23,14 @@ const Endpoints = {
     SurveySubmission: {
         CREATE: `${AppConfig.apiUrl}/submission/`,
     },
+    Comment: {
+        GET_ALL: `${AppConfig.apiUrl}/comments/`,
+        CREATE: `${AppConfig.apiUrl}/comments/`,
+        UPDATE: `${AppConfig.apiUrl}/comments/`,
+        GET: `${AppConfig.apiUrl}/comments/comment_id`,
+        LINK_TO_ENGAGEMENT: `${AppConfig.apiUrl}/comments/comment_id/link/engagement/engagement_id`,
```
Comment (djnunez-aot) [inline met-web/src/apiManager/endpoints/index.ts:31]: clean up

File: met-web/src/models/comment.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+export interface Comment {
+    id: number;
+    email: string;
+    comment_date: string;
+    published_date: string;
+    status: CommentStatus;
+    content: string;
+    reviewed_by: string;
+    date_reviewed: string;
```
Comment (djnunez-aot) [inline met-web/src/models/comment.ts:11]: surveyid

**Review by jadmsaadaot — APPROVED** [review]

**Review by VineetBala-AOT — APPROVED** [review]

**Pull request merged**